test/boost/total_order_check.hh

```diff
@@ -22,23 +22,27 @@
 #pragma once
 
 #include <vector>
+#include <compare>
 #include <boost/test/unit_test.hpp>
 #include <seastar/util/variant_utils.hh>
 #include <variant>
 #include "test/lib/log.hh"
+#include "utils/collection-concepts.hh"
 
 template<typename Comparator, typename... T>
 class total_order_check {
     using element = std::variant<T...>;
     std::vector<std::vector<element>> _data;
     Comparator _cmp;
 private:
-    static int sgn(int x) {
+    static int sgn(std::strong_ordering x) {
         return (x > 0) - (x < 0);
     }
-
+    static int sgn(int x) {
+        return sgn(x <=> 0);
```

Comment:
Wouldn't just "return x <=> 0" work? Why do you need to call sgn() on it? Maybe I'm missing something :-(

Reply:
Yes. The return value is int, not std::strong_ordering.